easybuild/easyblocks/generic/configuremake.py

```diff
@@ -83,11 +83,16 @@ def configure_step(self, cmd_prefix=''):
         if prefix_opt is None:
             prefix_opt = '--prefix='
 
-        cmd = "%(preconfigopts)s %(cmd_prefix)s./configure %(prefix_opt)s%(installdir)s %(configopts)s" % {
+        # Avoid using config.guess from the package as it is frequently out of date, use the version shipped with EB
+        build_type, _ = run_cmd('config_guess')
+        build_type_option = '--build=' + build_type.strip()
```

Comment:
@ocaisa As indicated by @micaeljtoliveira, this should be `config.guess`.
Also, please log what the result of `config_guess` is, and maybe also which `config.guess` is being (result of `which('config.guess')`).

Can we detect when `--build` needs to be used?
I'm a bit worried about the impact this may have on systems where this isn't needed currently...

Reply:
I don't know enough about Autotools to answer this... @geimer ?

I would expect that it does no harm, `config.guess` is intended to be updated and so backwards compatible, it's output is the default setting for `--build` (see https://www.gnu.org/software/autoconf/manual/autoconf-2.61/html_node/Specifying-Names.html).